src/Paket.Core/Installation/DependencyChangeDetection.fs

```diff
@@ -19,10 +19,10 @@ type DependencyChangeType =
 
 let findNuGetChangesInDependenciesFile(dependenciesFile:DependenciesFile,lockFile:LockFile,strict) =
     let allTransitives groupName = lockFile.GetTransitiveDependencies groupName
-    let getChanges groupName transitives (newRequirement:PackageRequirement) (originalPackage:ResolvedPackage) =
+    let getChanges groupName transitives (newRequirement:PackageRequirement) (originalPackage:PackageInfo) =
         let settingsChanged() =
             if newRequirement.Settings <> originalPackage.Settings then
-                if newRequirement.Settings = { originalPackage.Settings with FrameworkRestrictions = AutoDetectFramework } then
+                if newRequirement.Settings = { originalPackage.Resolved.Settings with FrameworkRestrictions = AutoDetectFramework } then
```

Comment:
originalPackage.Resolved.Settings doesn't combine with GroupSettings , here we need compare with the combined one originalPackage.Settings.

since like https://github.com/fsprojects/Paket/blob/master/src/Paket.Core/Versioning/Requirements.fs#L920-L925 here, we will get a None as the resolved settings's StorageConfig. But we will get a Some NoPackagesFolder from the newRequirement.Settings since newRequirement'settings does combine with group's settings :

   - https://github.com/fsprojects/Paket/blob/master/src/Paket.Core/Installation/DependencyChangeDetection.fs#L67
   - https://github.com/fsprojects/Paket/blob/master/src/Paket.Core/Installation/DependencyChangeDetection.fs#L80

Reply:
Can you please explain what the impact is (and send a test/pr as mentioned in the commit)?

Reply:
@matthid  #3464  here